### PR TITLE
Adds unit test for SiderMenuWrapper Component

### LINKED ISSUE #1337: dashboard components require further unit testing code coverage
In order to further improve the automated unit testing workflow for pbench dashboard, greater code coverage of dashboard components is required for catching all possible scenarios. By striving for complete code coverage, we can regularly maintain quality of tests, refactor components with confidence, and catch errors before deploying to production. 

Coverage reports are currently retrieved in Jest by running the following command: `umi test ./src/components ./src/pages --verbose --maxWorkers=1 --runInBand "--coverage"`

```
---------------------------------|----------|----------|----------|----------|-------------------|
File                             |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
---------------------------------|----------|----------|----------|----------|-------------------|
All files                        |    18.87 |     4.99 |     9.82 |     18.9 |                   |
 components/Button               |    33.33 |      100 |        0 |    33.33 |                   |
  index.js                       |    33.33 |      100 |        0 |    33.33 |               5,7 |
 components/MonthSelect          |       20 |        0 |        0 |       20 |                   |
  index.js                       |       20 |        0 |        0 |       20 |... 42,44,53,54,59 |
 components/PageHeader           |     7.35 |     2.63 |     5.88 |     7.46 |                   |
  index.js                       |     7.35 |     2.63 |     5.88 |     7.46 |... 11,218,225,250 |
 components/RowSelection         |       40 |        0 |        0 |       40 |                   |
  index.js                       |       40 |        0 |        0 |       40 |          19,20,22 |
 components/SearchBar            |    15.79 |        0 |        0 |    15.79 |                   |
  index.js                       |    15.79 |        0 |        0 |    15.79 |... 40,42,54,55,62 |
 components/SiderMenu            |    16.25 |     4.26 |       20 |    15.79 |                   |
  SiderMenu.js                   |    16.25 |     4.26 |       20 |    15.79 |... 13,214,215,217 |
 components/Table                |    33.33 |      100 |        0 |    33.33 |                   |
  index.js                       |    33.33 |      100 |        0 |    33.33 |               5,7 |
 components/TableFilterSelection |     5.26 |        0 |        0 |     5.41 |                   |
  index.js                       |     5.26 |        0 |        0 |     5.41 |... 97,106,109,131 |
 components/TimeseriesGraph      |    18.18 |        0 |        0 |    18.18 |                   |
  index.js                       |    18.18 |        0 |        0 |    18.18 |... 48,50,55,66,68 |
 components/_utils               |      100 |      100 |      100 |      100 |                   |
  pathTools.js                   |      100 |      100 |      100 |      100 |                   |
 layouts                         |        0 |        0 |        0 |        0 |                   |
  PageHeaderLayout.js            |        0 |        0 |        0 |        0 |                 7 |
 pages/ComparisonSelect          |    23.91 |        0 |    11.11 |    23.91 |                   |
  index.js                       |    23.91 |        0 |    11.11 |    23.91 |... 38,140,142,167 |
 pages/Controllers               |    26.67 |        0 |     9.09 |    26.67 |                   |
  index.js                       |    26.67 |        0 |     9.09 |    26.67 |... 46,152,176,178 |
 pages/Results                   |    26.19 |        0 |    11.76 |    26.19 |                   |
  index.js                       |    26.19 |        0 |    11.76 |    26.19 |... 37,148,177,179 |
 pages/RunComparison             |    23.01 |     7.69 |     6.98 |    23.21 |                   |
  index.js                       |    23.01 |     7.69 |     6.98 |    23.21 |... 65,568,601,603 |
 pages/Search                    |    27.78 |     8.33 |    11.11 |    27.78 |                   |
  index.js                       |    27.78 |     8.33 |    11.11 |    27.78 |... 69,272,303,305 |
 pages/Summary                   |    40.74 |    46.15 |    18.18 |    40.74 |                   |
  index.js                       |    40.74 |    46.15 |    18.18 |    40.74 |... 62,176,190,223 |
 utils                           |    14.15 |     3.52 |     7.46 |    14.47 |                   |
  parse.js                       |    19.28 |     5.56 |    11.11 |    19.27 |... 79,489,496,503 |
  utils.js                       |     2.94 |        0 |        0 |     3.23 |... 90,191,193,195 |
---------------------------------|----------|----------|----------|----------|-------------------|
```

### PR DESCRIPTION
**Pull request by Anisha1234**
Fixes #1337 
Code Coverage Report:

```
 File              |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line 
  index.js         |      100 |       75 |      100 |      100 |            21

```
Coverage reports are currently retrieved in Jest by running the following command: umi test ./src/components ./src/pages --verbose --maxWorkers=1 --runInBand "--coverage"

### CODE REVIEW

**Comment by Anisha1234** [comment]
@gurbirkalsi can you give some insights about how to test the conditional statement inside a component i.e `<SiderMenu {...props} collapsed={isMobile ? false : collapsed} />` 
I am not getting code coverage in only this line.

**Comment by gurbirkalsi** [comment]
@Anisha1234 we can apply tests for the ternary operator `?` by mocking the `collapsed` prop and check for both bool values `true` and `false`. For example:
```JavaScript
expect(wrapper.find(SiderMenu).prop('collapsed')).toBe(true)
expect(wrapper.find(SiderMenu).prop('collapsed')).toBe(false)
```

**Comment by Anisha1234** [comment]
@gurbirkalsi I tried doing the exact thing but it didn't enhance the code coverage. I am still getting uncovered branch at the same point.

**Comment by portante** [comment]
@Anisha1234, where are we at with this change?  Can you rebase this change, and then show how you are mocking out the `collapsed` prop?

File: web-server/v0.4/src/components/SiderMenu/index.test.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      .first()
+      .props()
+      .onMaskClick();
+    expect(onCollapse).toHaveBeenCalledTimes(2);
+  });
+  it('test the Sider Menu functions', () => {
+    wrapper.setProps({ isMobile: true });
+    expect(wrapper.find(DrawerMenu).find(SiderMenu)).toHaveLength(1);
+  });
+  it('test the Sider Menu collapsing', () => {
```
Comment (Anisha1234) [inline web-server/v0.4/src/components/SiderMenu/index.test.js]: @portante Here I am mocking the collapse prop by varying the isMobile prop. I have also rebased the PR.

**Pull request merged**